source/extensions/common/aws/signer_base_impl.cc

```diff
@@ -124,13 +125,24 @@ void SignerBaseImpl::sign(Http::RequestHeaderMap& headers, const std::string& co
     // Append signature to existing query string
     query_params.add(SignatureQueryParameters::get().AmzSignature, signature);
     headers.setPath(query_params.replaceQueryString(headers.Path()->value()));
-    ENVOY_LOG(debug, "Query string appended to path");
+    // Sanitize logged query string
+    query_params.overwrite(SignatureQueryParameters::get().AmzSignature, "*****");
+    query_params.overwrite(SignatureQueryParameters::get().AmzSecurityToken, "*****");
+    auto sanitised_query_string = query_params.replaceQueryString(headers.Path()->value());
+    ENVOY_LOG(debug, "Query string signing - New path (sanitised): {}", sanitised_query_string);
+
   } else {
     const auto authorization_header = createAuthorizationHeader(
         credentials.accessKeyId().value(), credential_scope, canonical_headers, signature);
 
-    ENVOY_LOG(debug, "Signing request with: {}", authorization_header);
     headers.addCopy(Http::CustomHeaders::get().Authorization, authorization_header);
+    Envoy::Logger::Registry::setLogLevel(spdlog::level::debug);
+
+    // Sanitize logged authorization header
+    std::vector<std::string> sanitised_header =
+        absl::StrSplit(authorization_header, absl::ByString("Signature="));
+    ENVOY_LOG(debug, "Header signing - Authorization header (sanitised): {}Signature=*****",
+              sanitised_header[0]);
```

Comment:
if possible would you show an example of what this looks like before and after?

Reply:
After: 
Header signing - Authorization header (sanitised): AWS4-HMAC-SHA256 Credential=ASIAQ1234567/20240223/ap-southeast-2/vpc-lattice-svcs/aws4_request, SignedHeaders=accept;host;user-agent;x-amz-content-sha256;x-amz-date;x-amz-security-token;x-request-id, Signature=****